lib/ansible/plugins/callback/foreman.py

```diff
@@ -80,154 +80,173 @@
 from ansible.plugins.callback import CallbackBase
 
 
+def build_log(data):
+    """
+    Transform the internal log structure to one accepted by Foreman's
+    config_report API.
+    """
+    for source, msg in data:
+        if 'failed' in msg:
+            level = 'err'
+        elif 'changed' in msg and msg['changed']:
+            level = 'notice'
+        else:
+            level = 'info'
+
+        yield {
+            "log": {
+                'sources': {
+                    'source': source,
+                },
+                'messages': {
+                    'message': json.dumps(msg),
+                },
+                'level': level,
+            }
+        }
+
+
+def get_time():
+    """
+    Return the time for measuring duration. Prefers monotonic time but
+    falls back to the regular time on older Python versions.
+    """
+    try:
+        return time.monotonic()
+    except AttributeError:
+        return time.time()
+
+def get_now():
+    """
+    Return the current timestamp as a string to be sent over the network.
+    """
+    return datetime.utcnow().isoformat()
+
+
 class CallbackModule(CallbackBase):
     CALLBACK_VERSION = 2.0
     CALLBACK_TYPE = 'notification'
     CALLBACK_NAME = 'foreman'
     CALLBACK_NEEDS_WHITELIST = True
 
-    FOREMAN_HEADERS = {
-        "Content-Type": "application/json",
-        "Accept": "application/json"
-    }
-    TIME_FORMAT = "%Y-%m-%d %H:%M:%S %f"
-
     def __init__(self):
         super(CallbackModule, self).__init__()
         self.items = defaultdict(list)
-        self.start_time = int(time.time())
+        self.facts = defaultdict(dict)
+        self.start_time = get_time()
 
     def set_options(self, task_keys=None, var_options=None, direct=None):
 
         super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
 
-        self.FOREMAN_URL = self.get_option('url')
-        self.FOREMAN_SSL_CERT = (self.get_option('client_cert'), self.get_option('client_key'))
-        self.FOREMAN_SSL_VERIFY = str(self.get_option('verify_certs'))
-
-        self.ssl_verify = self._ssl_verify()
+        self.foreman_url = self.get_option('url')
+        ssl_cert = self.get_option('client_cert')
+        ssl_key = self.get_option('client_key')
 
         if HAS_REQUESTS:
-            requests_major = int(requests.__version__.split('.')[0])
-            if requests_major < 2:
-                self._disable_plugin('The `requests` python module is too old.')
+            requests_version = requests.__version__.split('.')
+            if int(requests_version[0]) < 2 or int(requests_version[1]) < 14:
```

Comment:
And then this check would be way more readable:
```suggestion
            if requests_version < (2, 14):
```